Gems/Meshlets/Code/Source/Meshlets/MeshletsRenderObject.h

```diff
@@ -0,0 +1,193 @@
+
+        class MeshletsFeatureProcessor;
+
+        // The following structure holds the per object data - currently with no support
+        // for instancing.
+        // To support instancing move the dispatch item, draw packet and object Id to
+        // a separate instance data structure.
+        // The data here should only represent the object render/compute data without
+        // having any instance data (matrices, Id, etc..)
+        struct MeshRenderData
+        {
+            Render::TransformServiceFeatureProcessorInterface::ObjectId ObjectId;   // should be per instance
+
+            uint32_t MeshletsCount = 0;
+
+            // Used by the direct Draw stage only - should be changed for indirect culled render.
+            uint32_t IndexCount = 0;  
+
+             //! Compute render data
+            Data::Instance<RPI::ShaderResourceGroup> ComputeSrg;          // Per object Compute data - can be shared across instances
+            AZStd::vector<SrgBufferDescriptor> ComputeBuffersDescriptors;
+            AZStd::vector<Data::Instance<RHI::BufferView>> ComputeBuffersViews;
+            AZStd::vector<Data::Instance<Meshlets::SharedBufferAllocation>> ComputeBuffersAllocators;
+            AZStd::vector <Data::Instance<RPI::Buffer>> ComputeBuffers;   // stand alone non shared buffers
+            MeshletsDispatchItem MeshDispatchItem;
+
+            //! Render pass data
+            Data::Instance<RPI::ShaderResourceGroup> RenderObjectSrg;     // Per object render data - includes instanceId and vertex buffers
+            AZStd::vector<SrgBufferDescriptor> RenderBuffersDescriptors;
+            RHI::IndexBufferView IndexBufferView;
+            AZStd::vector<Data::Instance<RHI::BufferView>> RenderBuffersViews; 
+            AZStd::vector <Data::Instance<RPI::Buffer>> RenderBuffers;    // stand alone non shared buffers
+
+            const RHI::DrawPacket* MeshDrawPacket = nullptr;    // Should be moved to the instance data structure
+        };
+        using ModelLodDataArray = AZStd::vector<MeshRenderData*>;    // MeshRenderData per mesh in the Lod
+
+        //! Currently assuming single model without Lods so that the handling of the
+        //! meshlet creation and handling of the array is easier. If several meshes or Lods
+        //! exist, they will be created as separate models and the last model's instance
+        //! will be kept in this class.
+        //! To enhance this, add inheritance to lower levels of the model / mesh.
+        //! MeshletsModel represents a combined model that can contain an array
+        //! of ModelLods.
+        //! Each one of the ModelLods contains a vector of meshes, representing possible multiple
+        //! element within the mesh.
+        class MeshletsRenderObject
+        {
+        public:
+            static uint32_t s_modelNumber;
+
+            Name s_TextureCoordinatesName;
+            Name s_IndicesName;
```

Comment:
nit: I think statics are still supposed to use s_lowerCamelCase

Reply:
absolutely - changed